### PR TITLE
Add basicapitest.cpp for GitHub Actions CI

### PR DESCRIPTION
**Pull request by Shreeshrii**

### CODE REVIEW

File: testing/basicapitest.cpp [hunk 1]
```diff
@@ -0,0 +1,29 @@
+#include <tesseract/baseapi.h>
+#include <leptonica/allheaders.h>
```
Comment (stweil) [inline testing/basicapitest.cpp:2]:
```suggestion
#include <allheaders.h>
```

File: testing/basicapitest.cpp [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+
+int main()
+{
+    char *outText;
+
```
Comment (stweil) [inline testing/basicapitest.cpp:7]:
```suggestion
```

File: testing/basicapitest.cpp [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+    if (api->Init("../../tessdata", "eng")) {
+        fprintf(stderr, "Could not initialize tesseract.\n");
+        exit(1);
+    }
+
+    // Open input image with leptonica library
+    Pix *image = pixRead("../test/testing/phototest.tif");
+    api->SetImage(image);
+    // Get OCR result
+    outText = api->GetUTF8Text();
```
Comment (stweil) [inline testing/basicapitest.cpp:19]:
```suggestion
    char *outText = api->GetUTF8Text();
```

File: testing/basicapitest.cpp (hunk 2 above)
Comment (stweil) [reply testing/basicapitest.cpp:7]: I'd also add `#include <stdio.h>` to be on the safe side.

**Comment by Shreeshrii** [comment]
Thanks. I have made the suggested changes.

File: testing/basicapitest.cpp [hunk 4]
```diff
@@ -0,0 +1,29 @@
+#include <allheaders.h>
+#include <stdio.h>
+#include <tesseract/baseapi.h>
+
+int main()
+{
+
```
Comment (stweil) [inline testing/basicapitest.cpp:7]:
```suggestion
```

**Comment by stweil** [comment]
Thanks. I have also updated the `test` submodule for Tesseract (commit https://github.com/tesseract-ocr/tesseract/commit/3d5fea6355be06052b9b9ed482693d27c26cf892).

**Pull request merged**